File: livedatademo/src/main/java/com/example/livedatademo/MainActivity.java
package com.example.livedatademo;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;

import androidx.appcompat.app.AppCompatActivity;

import com.example.livedatademo.databus.LiveDataBus;


public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
    }

    public void click(View view) {
        if (view.getId() == R.id.btn) {
            Intent intent = new Intent(this, NameActivity.class);
            startActivity(intent);
        } else if (view.getId() == R.id.btn2) {
            Intent intent = new Intent(this, TestLiveDataBusActivity.class);
            startActivity(intent);

            new Thread(){
                @Override
                public void run() {
                    for (int i = 0; i < 10; i++) {
                        //发送消息
                        LiveDataBus.getInstance().with("data", String.class).postValue("jett");
                        try {
                            Thread.sleep(5000);
                        } catch (InterruptedException e) {
                            e.printStackTrace();
                        }


                    }
                }
            }.start();


        }
    }


}

File: customizeview/src/main/java/com/example/customizeview/MainActivity.kt
package com.example.customizeview

import android.content.Intent
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import com.example.customizeview.slidecardview2.SlideCardActivity
import com.example.customizeview.toprecycleview.TopRecycleViewActivity
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        tv_to_viewpager.setOnClickListener {
            val intent: Intent = Intent(this@MainActivity, ViewpagerDemoActivity::class.java)
            startActivity(intent)
        }

        tv_to_center_text.setOnClickListener {
            val intent = Intent(this@MainActivity, CenterTextActivity::class.java)
            startActivity(intent)
        }

        tv_to_fish.setOnClickListener {
            val intent: Intent = Intent(this@MainActivity, FishActivity::class.java)
            startActivity(intent)
        }

        tv_to_recycle_view.setOnClickListener {
            val intent: Intent = Intent(this@MainActivity, TopRecycleViewActivity::class.java)
            startActivity(intent)
        }

        tv_to_slide_card_view.setOnClickListener{
            val intent: Intent = Intent(this@MainActivity, SlideCardActivity::class.java)
            startActivity(intent)
        }
    }
}
